rpc/src/client.rs

```diff
@@ -1,31 +1,45 @@
 //! Tendermint RPC client
 
-use bytes::buf::ext::BufExt;
-use hyper::header;
+use tendermint::{
+    abci::{self, Transaction},
+    block::Height,
+    evidence::Evidence,
+    net, Genesis,
+};
 
-use tendermint::abci::{self, Transaction};
-use tendermint::block::Height;
-use tendermint::evidence::Evidence;
-use tendermint::net;
-use tendermint::Genesis;
-
-use crate::{endpoint::*, Error, Request, Response};
+use crate::{
+    client::{
+        subscription::SubscriptionManager,
+        transport::{http_ws::HttpWsTransport, Transport},
+    },
+    endpoint::*,
+    Error, Request, Response,
+};
 
 pub mod event_listener;
+pub mod subscription;
+pub mod transport;
+
+#[cfg(test)]
+pub mod testing;
 
 /// Tendermint RPC client.
 ///
 /// Presently supports JSONRPC via HTTP.
-#[derive(Clone, Debug)]
+#[derive(Debug)]
```

Comment:
What prevents us to keep `Clone` on the client?

Reply:
We'd need to be able to clone the transport (trait object), which I thought may not make sense for all possible transports.

For example, what would the cloning implications be for a purely WebSocket-based client? Would it be useful to clone such a client, and what would it mean to do so? (i.e. do we open a new connection, or do we provide another interface to the same underlying connection?)